src/main/scala/treadle/executable/BlackBoxCycler.scala

```diff
@@ -5,6 +5,16 @@ package treadle.executable
 import firrtl.ir.Info
 import treadle.ScalaBlackBox
 
+/**
+  * Implements an assigner that can be scheduled to publish clock transitions
+  * to specific black box implementations
+  * @param symbol symbol name of instance
+  * @param blackBox the instance
+  * @param clockSymbol clock used by instance
+  * @param dataStore the data
+  * @param info source location
+  */
+//TODO: This should be called everytime something hanppens to clock and should indicate all possible transitions
```

Comment:
Is it the `run` method that should be called? It would be useful to have an example of how to use this from client code.

Reply:
All `Assigner`s have a run method in order to update the wire they are associated with, BlackBox clocks use this *magic* assigner, who's run method is only called when the associated clock changes. `run` is an internal API and not documented itself, but by convention is documented (usually :-) in the documentation of the Assigner class, as it is here.